### DISCUSSION TITLE
Plug-In contribution

### DISCUSSION THREAD

**Discussion by felipecosta09** [trendmicro/cloudone-filestorage-plugins#97] [Q&A] [answered]
I have a genuine question about how to make contributions in the post-scan section.

If we have a plugin already in the repo using CloudFormation as a deployment method or python as a programming language and I want to provide a different deployment method (Like Terraform) or programming language (Like NodeJS)...

Should I create another folder or should add that "feature" to the current plugin?

**Comment by carpusherw** [comment, marked as answer]
For programing language, it is definitely a different plugin/folder as we already design the language infix in the names.
Other deployment method is open to discussion. A precedent we have is [aws-python-promote-or-quarantine](../tree/master/post-scan-actions/aws-python-promote-or-quarantine), which has a [other-installation-methods.md](../blob/master/post-scan-actions/aws-python-promote-or-quarantine/other-installation-methods.md) that describes a manual installation method in addition to the default SAR approach. The story behind is that we used the SAR approach to deprecate the manual.
